kyaml/fn/framework/command/doc.go

```diff
@@ -0,0 +1,87 @@
+// Package command contains a builder for creating cobra.Commands based on configuration functions
+// written using the kyaml function framework. The commands this package generates can be used as
+// standalone executables or as part of a a configuration management pipeline that complies with the
+// Configuration Functions Specification (e.g. Kustomize generators or transformers):
+// https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/api-conventions/functions-spec.md
+//
+// Example standalone usage
+//
+// Function template input:
+//
+//    # config.yaml -- this is the input to the template
+//    apiVersion: example.com/v1alpha1
+//    kind: Example
+//    Key: a
+//    Value: b
+//
+// Additional function inputs:
+//
+//    # patch.yaml -- this will be applied as a patch
+//    apiVersion: apps/v1
+//    kind: Deployment
+//    metadata:
+//      name: foo
+//      namespace: default
+//      annotations:
+//        patch-key: patch-value
+//
+// Manually run the function:
+//
+//    # build the function
+//    $ go build example-fn/
+//
+//    # run the function
+//    $ ./example-fn config.yaml patch.yaml
+//
+// Go implementation
+//
+//   // example-fn/main.go
+//	func main() {
+//		// Define the template used to generate resources
+//		p := framework.TemplateProcessor{
+//			MergeResources: true, // apply inputs as patches to the template output
+//			TemplateData: &struct {
```

Comment:
What's the point of passing this `TemplateData` if it's all empty? Would it be different than to pass nil or not to pass anything?

Also, I would definitely use `new(struct { ... })` in this case, if you have to

Reply:
It's empty now, because the framework populates its data during processing from the input ResourceList! So the data in this example is in `config.yaml`. If you did not pass this, the ResourceList.FunctionConfig would not get loaded anywhere, and the templates would not have access to its data.